payment-service/src/main/java/com/example/paymentservice/config/logging/LoggingAspect.java

```diff
@@ -63,22 +73,77 @@ public void logAfterThrowing(JoinPoint joinPoint, Throwable e) {
 
     @Around("applicationPackagePointcut()")
     public Object logAround(ProceedingJoinPoint joinPoint) throws Throwable {
-        if (log.isDebugEnabled()) {
-            log.debug(
-                    "Enter: {}.{}()",
-                    joinPoint.getSignature().getDeclaringTypeName(),
-                    joinPoint.getSignature().getName());
-        }
+        LogLevel logLevel = determineLogLevel(joinPoint);
+        String methodName = joinPoint.getSignature().getName();
+
+        logWhenEnabled(joinPoint, LogLevel.INFO, methodName + "() start execution");
+        logMethodParamsIfEnabled(joinPoint, logLevel, methodName);
+
         long start = System.currentTimeMillis();
         Object result = joinPoint.proceed();
         long end = System.currentTimeMillis();
-        if (log.isDebugEnabled()) {
-            log.debug(
-                    "Exit: {}.{}(). Time taken: {} millis",
-                    joinPoint.getSignature().getDeclaringTypeName(),
-                    joinPoint.getSignature().getName(),
-                    end - start);
-        }
+
+        logMethodResultIfEnabled(joinPoint, result, logLevel, methodName);
+
+        logWhenEnabled(
+                joinPoint,
+                LogLevel.INFO,
+                methodName
+                        + "() finished execution and took ("
+                        + +(end - start)
+                        + ") mills to execute");
+
         return result;
     }
+
+    private LogLevel determineLogLevel(ProceedingJoinPoint joinPoint) {
+        MethodSignature methodSignature = (MethodSignature) joinPoint.getSignature();
```

Comment:
@coderabbitai, code from line 100 to 104 and 109 to 115 looks similar, can you please refactor the code

Reply:
> @rajadilipkolli, to address the redundancy between the code blocks from lines 100 to 104 and 109 to 115 in the `LoggingAspect.java` file, I recommend encapsulating the logging logic into a private method. This method can then be called from both places with the appropriate parameters. Here's how the refactored code could look:
> 
> ```java
> // Refactored method to encapsulate logging logic
> private void logExecutionDetails(ProceedingJoinPoint joinPoint, LogLevel logLevel, String message) {
>     logWhenEnabled(joinPoint, logLevel, message);
> }
> 
> @Around("applicationPackagePointcut()")
> public Object logAround(ProceedingJoinPoint joinPoint) throws Throwable {
>     LogLevel logLevel = determineLogLevel(joinPoint);
>     String methodName = joinPoint.getSignature().getName();
> 
>     // Log start of method execution
>     logExecutionDetails(joinPoint, LogLevel.INFO, methodName + "() start execution");
>     logMethodParamsIfEnabled(joinPoint, logLevel, methodName);
> 
>     long start = System.currentTimeMillis();
>     Object result = joinPoint.proceed();
>     long end = System.currentTimeMillis();
> 
>     logMethodResultIfEnabled(joinPoint, result, logLevel, methodName);
> 
>     // Log completion of method execution
>     logExecutionDetails(
>             joinPoint,
>             LogLevel.INFO,
>             methodName
>                     + "() finished execution and took ("
>                     + (end - start)
>                     + ") mills to execute");
> 
>     return result;
> }
> ```
> 
> This refactoring introduces a `logExecutionDetails` method that encapsulates the logging logic, reducing redundancy and improving code maintainability.

@coderabbitai, can you please recheck the lines, if you can't determine I can paste both methods.

Reply:
@coderabbitai, below both methods has similar body of logic, except that return type is different, please refactor from above code perspective 

```java
private LogLevel determineLogLevel(ProceedingJoinPoint joinPoint) {
        MethodSignature methodSignature = (MethodSignature) joinPoint.getSignature();
        Method method = methodSignature.getMethod();
        Loggable methodAnnotation = method.getAnnotation(Loggable.class);
        Loggable classAnnotation = joinPoint.getTarget().getClass().getAnnotation(Loggable.class);
        return Optional.ofNullable(methodAnnotation).orElse(classAnnotation).value();
    }

    private boolean shouldLog(
            ProceedingJoinPoint joinPoint, Function<Loggable, Boolean> loggableProperty) {
        Loggable methodAnnotation =
                ((MethodSignature) joinPoint.getSignature())
                        .getMethod()
                        .getAnnotation(Loggable.class);
        Loggable classAnnotation = joinPoint.getTarget().getClass().getAnnotation(Loggable.class);
        return loggableProperty.apply(
                Optional.ofNullable(methodAnnotation).orElse(classAnnotation));
    }
```

Reply:
@coderabbitai, last line in the code should be 

```java
return Optional.ofNullable(Optional.ofNullable(methodAnnotation).orElse(classAnnotation));
```

After changing the line, it is working, hence will mark as resolved.